2_process/src/munge_inst_timeseries.R

```diff
@@ -55,3 +55,36 @@ aggregate_data_to_hourly <- function(inst_data,output_tz){
 
 }
 
+aggregate_data_to_daily <- function(inst_data, daily_data, output_tz){
+  #' 
+  #' @description Function to aggregate instantaneous NWIS data collected at sub-hourly (e.g. 15/5/30 min) intervals to hourly min/mean/maxs
+  #'
+  #' @param inst_data a data frame containing the downloaded time series for NWIS instantaneous site. 
+  #' inst_data must include the following columns: c("Value_Inst","dateTime","site_no")
+  #' @param daily_data a data frame the downloaded daily time series of DO data. This is used so that if a site is already in 
+  #'  the daily sites, we won't do the aggregating here
+  #' @param output_tz character string to set display attribute of dateTime. Possible values to provide are "UTC",
+  #' "America/New_York","America/Chicago", "America/Denver","America/Los_Angeles", "America/Anchorage", 
+  #' as well as the following which do not use daylight savings time: "America/Honolulu", "America/Jamaica",
+  #' "America/Managua","America/Phoenix", and "America/Metlakatla"
+  #'
+  #' @value A data frame containing daily min, mean, and max values the parameter of interest
+  
+  
+  only_inst_data = setdiff(inst_data$site_no, daily_data$site_no)
+  
+  daily_values <- inst_data %>%
+    filter(site_no %in% only_inst_data) %>%
+    mutate(dateTime = with_tz(dateTime,tzone=output_tz)) %>%
+    mutate(Date = date(dateTime)) %>%
+    group_by(Date, site_no) %>%
+    summarise(Value = mean(Value_Inst, na.rm=TRUE), 
+              Value_Min = min(Value_Inst, na.rm=TRUE), 
+              Value_Max = max(Value_Inst,na.rm=TRUE), 
```

Comment:
@jsadler2 I just tried running the code and ran into warnings when all values for a day are NA. Do you have that problem, too?

This can happen if a site reports multiple time series/sensor locations within the river and one of those begins before the either of the time series we select. I see this in site 01434498, for example. One way to get around this problem is to calculate `na_count`, `value_count`, and `percent_coverage` and filter by percent_coverage before calculating the mean/max/min.

Reply:
I realized I got this error when running and testing the daily aggregation steps on the full `p1_inst_data`. So perhaps this isn't an issue for the data subset returned by the `setdiff` step.